pkg/repo/migration.go

```diff
@@ -34,9 +34,12 @@ func (m *MigrationManager) Add(migration Migration) error {
 
 // Migrate runs the migrations on the given repo.
 func (m *MigrationManager) Migrate(repo FsRepo) error {
-	currentVersion, err := repo.readVersion()
+	currentVersion, err := repo.readLegacyVersion()
 	if err != nil {
-		return err
+		currentVersion, err = repo.readVersion()
+		if err != nil {
+			return err
+		}
 	}
```

Comment:
I am looking at `repo.readVersion()` implementation, and it falls back to `readLegacyVersion()` if no version was read. Should we just keep the fallback logic inside `readVersion()` instead of repeating it here as well?

Reply:
Yes good idea, will adjust.